src/controllers/followController.js

```diff
@@ -0,0 +1,206 @@
+import Models from '../db/models';
+import errorResponse from '../helpers/index';
+
+const { User, Follow } = Models;
+
+/**
+ * @description handles follow and unfollow
+ * @class FollowController
+ */
+class FollowController {
+  /**
+   * @description Handles the follow feature
+   * @param {object} request - Request object
+   * @param {object} response - Response object
+   * @returns {object} An object containing all the data related to the unfollowed user
+   * @memberof FollowController
+   */
+  static async followUser(request, response) {
+    const { username } = request.params;
+    const followingId = request.userData.payload.id;
+    const user = await User.findOne({
+      where: {
+        username
+      }
+    });
+    try {
+      if (user) {
+        if (followingId !== user.id) {
+          Follow.findOrCreate({
+            where: {
+              userId: user.id,
+              followerId: followingId
+            }
+          }).spread((check, created) => {
+            if (!created) {
+              return response.status(409).json(errorResponse([`You are already following ${username}`]));
+            }
+            return response.status(200).json({
+              message: `You are now following ${username}`
+            });
+          });
+        } else {
+          return response.status(409).json(errorResponse(['You cannot follow yourself']));
+        }
+      } else {
+        return response.status(404).json(errorResponse(['User not found']));
+      }
+    } catch (error) {
+      return response.status(500).json(errorResponse([error.message]));
+    }
+  }
+
+  /**
+   * @description Handles the unfollow feature
+   * @param {object} request - Request object
+   * @param {object} response - Response object
+   * @returns {object} An object containing all the data related to the unfollowed user
+   * @memberof FollowController
+   */
+  static async unfollowUser(request, response) {
+    const { username } = request.params;
+    const user = await User.findOne({
+      where: {
+        username
+      }
+    });
+    try {
+      if (user) {
+        const followingId = request.userData.payload.id;
+        if (followingId !== user.id) {
+          const unfollow = await Follow.destroy({
```

Comment:
@frostyblok nice implementation with the unfollowing a user, however I feel destroying data might cause some discrepancies in the database table (I might be wrong), I feel just having an extra column with boolean values true/false would preserve the data rather than just deleting it. So when a user unfollows a user, the column will just have false and when a user follows another, it just has true. What do you think?